From 1c7665f8c6f037d8b63c9afa4c6b78089e27cf9a Mon Sep 17 00:00:00 2001
From: Rex Zhu <Rex.Zhu@amd.com>
Date: Thu, 17 Dec 2015 14:20:06 +0800
Subject: [PATCH 1349/1767] drm/amd/powerplay: check whether need to enable
 thermal control. (v2)

BugLink: http://bugs.launchpad.net/bugs/1546572

In I+A platform(skylake), it is controlled by intel.

v2: integrate Tom's fix

Reviewed-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Rex Zhu <Rex.Zhu@amd.com>
(cherry picked from commit cae9b9c81bde812590cdac7df32ad5662741b3d5)
Signed-off-by: Alberto Milone <alberto.milone@canonical.com>

Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 drivers/gpu/drm/amd/powerplay/eventmgr/eventtasks.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/powerplay/eventmgr/eventtasks.c b/drivers/gpu/drm/amd/powerplay/eventmgr/eventtasks.c
index 0a03f79..f0700d0 100644
--- a/drivers/gpu/drm/amd/powerplay/eventmgr/eventtasks.c
+++ b/drivers/gpu/drm/amd/powerplay/eventmgr/eventtasks.c
@@ -418,10 +418,17 @@ restart_search:
 int pem_task_initialize_thermal_controller(struct pp_eventmgr *eventmgr, struct pem_event_data *event_data)
 {
 	struct PP_TemperatureRange range;
+
 	range.max = TEMP_RANGE_MAX;
 	range.min = TEMP_RANGE_MIN;
 
-	return phm_start_thermal_controller(eventmgr->hwmgr, &range);
+	if (eventmgr == NULL || eventmgr->platform_descriptor == NULL)
+		return -EINVAL;
+
+	if (phm_cap_enabled(eventmgr->platform_descriptor->platformCaps, PHM_PlatformCaps_ThermalController))
+		return phm_start_thermal_controller(eventmgr->hwmgr, &range);
+
+	return 0;
 }
 
 int pem_task_uninitialize_thermal_controller(struct pp_eventmgr *eventmgr, struct pem_event_data *event_data)
-- 
1.9.3

